tests/test_utils.py

```diff
@@ -35,6 +36,85 @@ def test_buildconfig_update():
     assert x == {'a': 'A', 'strategy': {'b1': 'newB1', 'b3': 'B3', 'b11': {}}, 'c': 'C', 'd': 'D'}
 
 
+def has_ist(x):
+    if not has_triggers(x):
+        return False
+    triggers = x['spec']['triggers']
+    if not triggers:
+        return False
+    for t in triggers:
+        if t.get('type', None) == 'ImageChange':
+            return True
+    return False
+
+
+def has_other_trigger(x):
+    if not has_triggers(x):
+        return False
+    triggers = x['spec']['triggers']
+    if not triggers:
+        return False
+    for t in triggers:
+        if t.get('type', None) != 'ImageChange':
+            return True
+    return False
+
+
+def add_ist(x):
+    if 'triggers' not in x['spec']:
+        x['spec']['triggers'] = []
+    x['spec']['triggers'].append({
+        "type": "ImageChange",
+        "imageChange": {
+            "from": {
+                "kind": "ImageStreamTag",
+                "name": "foobar"
```

Comment:
I generally like having a trailing comma after every dict entry, even the last, as it makes diffs easier to read later if entries are added/removed.